hudi-common/src/main/java/org/apache/hudi/common/model/HoodieLogFile.java

```diff
@@ -80,27 +85,42 @@ public HoodieLogFile(String logPathStr) {
   }
 
   public String getFileId() {
-    return FSUtils.getFileIdFromLogPath(getPath());
+    if (fileId == null) {
+      this.fileId = FSUtils.getFileIdFromLogPath(getPath());
```

Comment:
Instead of running same regex for every component, let's run it once and set all of them at once